From e8f07821cce1bd0ab6d5622c2a42440f15f4fd71 Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Tue, 24 Dec 2013 21:53:23 +0100
Subject: [PATCH 01/34] 	* configure.ac: Set version to 2.02~beta2.

---
 ChangeLog    | 4 ++++
 configure.ac | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index 3545ea1..41bcebf 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,9 @@
 2013-12-24  Vladimir Serbinenko  <phcoder@gmail.com>
 
+	* configure.ac: Set version to 2.02~beta2.
+
+2013-12-24  Vladimir Serbinenko  <phcoder@gmail.com>
+
 	* grub-core/disk/efi/efidisk.c (name_devices): Skip Apple ghosts.
 
 2013-12-24  Andrey Borzenkov <arvidjaar@gmail.com>
diff --git a/configure.ac b/configure.ac
index 2481410..7c5d080 100644
--- a/configure.ac
+++ b/configure.ac
@@ -32,7 +32,7 @@ dnl type, so there is no conflict. Variables with the prefix "TARGET_"
 dnl (such as TARGET_CC, TARGET_CFLAGS, etc.) are used for the target
 dnl type.
 
-AC_INIT([GRUB],[2.02~beta1],[bug-grub@gnu.org])
+AC_INIT([GRUB],[2.02~beta2],[bug-grub@gnu.org])
 
 AC_CONFIG_AUX_DIR([build-aux])
 
-- 
1.8.4.2

