From ede6c9689395499f98a4c14d76b97151b4de5f54 Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Wed, 3 May 2017 12:58:15 +0200
Subject: [PATCH 011/194] Add strtoull test.

---
 grub-core/Makefile.core.def | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/grub-core/Makefile.core.def b/grub-core/Makefile.core.def
index 2dfa22a92..8dcd0e5a9 100644
--- a/grub-core/Makefile.core.def
+++ b/grub-core/Makefile.core.def
@@ -1962,6 +1962,11 @@ module = {
 };
 
 module = {
+  name = strtoull_test;
+  common = tests/strtoull_test.c;
+};
+
+module = {
   name = setjmp_test;
   common = tests/setjmp_test.c;
 };
-- 
2.13.5

