From 4f8471532d454b2db96434c28bf7105ed69e0f4d Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Tue, 9 May 2017 08:39:40 +0200
Subject: [PATCH 030/194] Fix bug on FDT nodes with compatible property

---
 grub-core/bus/fdt.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/grub-core/bus/fdt.c b/grub-core/bus/fdt.c
index 6fb077000..34d535457 100644
--- a/grub-core/bus/fdt.c
+++ b/grub-core/bus/fdt.c
@@ -43,6 +43,8 @@ is_compatible (struct grub_fdtbus_driver *driver,
   grub_size_t compatible_size;
   const char *compatible = grub_fdt_get_prop (dtb, node, "compatible",
 					      &compatible_size);
+  if (!compatible)
+    return 0;
   const char *compatible_end = compatible + compatible_size;
   while (compatible < compatible_end)
     {
-- 
2.13.5

