From cb8b7e4e36bf32dccbbba8252aa44ad3ada4fbc1 Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Sun, 9 Jul 2017 21:48:37 +0000
Subject: [PATCH 048/194] arm-efi: Fix compilation

---
 grub-core/loader/arm/linux.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/grub-core/loader/arm/linux.c b/grub-core/loader/arm/linux.c
index 260cbf068..e64c79a95 100644
--- a/grub-core/loader/arm/linux.c
+++ b/grub-core/loader/arm/linux.c
@@ -252,12 +252,12 @@ linux_boot (void)
 #ifdef GRUB_MACHINE_EFI
       grub_size_t size;
       if (fdt_valid)
-	size = grub_fdt_get_totalsize (fdt_addr);
+	size = grub_fdt_get_totalsize (current_fdt);
       else
-	size = 4 * get_atag_size (atag_orig);
+	size = 4 * get_atag_size (current_fdt);
       size += grub_strlen (linux_args) + 256;
       target_fdt = grub_efi_allocate_loader_memory (LINUX_FDT_PHYS_OFFSET, size);
-      if (!fdt_addr)
+      if (!target_fdt)
 	return grub_error (GRUB_ERR_OUT_OF_MEMORY, N_("out of memory"));
 #else
       target_fdt = (void *) LINUX_FDT_ADDRESS;
@@ -522,7 +522,7 @@ GRUB_MOD_INIT (linux)
 					  /* TRANSLATORS: DTB stands for device tree blob.  */
 					  0, N_("Load DTB file."));
   my_mod = mod;
-  current_fdt = grub_arm_firmware_get_boot_data ();
+  current_fdt = (const void *) grub_arm_firmware_get_boot_data ();
   machine_type = grub_arm_firmware_get_machine_type ();
 }
 
-- 
2.13.5

