From 68d54b55f49249c5e0517bf189f0f3df7ee6a36f Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Mon, 10 Jul 2017 01:34:22 +0000
Subject: [PATCH 049/194] fdt: silence clang warning.

---
 grub-core/lib/fdt.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/grub-core/lib/fdt.c b/grub-core/lib/fdt.c
index bdc630244..2705f2629 100644
--- a/grub-core/lib/fdt.c
+++ b/grub-core/lib/fdt.c
@@ -309,7 +309,8 @@ advance_token (const void *fdt, const grub_uint32_t *token, const grub_uint32_t
 	    continue;
 	  }
 	char *ptr;
-	for (ptr = (char *) (token + 1); *ptr && ptr < (char *) end; ptr++);
+	for (ptr = (char *) (token + 1); *ptr && ptr < (char *) end; ptr++)
+	  ;
         if (ptr >= (char *) end)
           return 0;
 	return token;
-- 
2.13.5

