From 2ccaa3b9fdc3e71e9e2e68acfa3054233a2a30b2 Mon Sep 17 00:00:00 2001
From: Andrei Borzenkov <arvidjaar@gmail.com>
Date: Thu, 20 Nov 2014 20:54:49 +0300
Subject: [PATCH 115/506] Fix date in last ChangeLog entry

---
 ChangeLog | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index 07d0646..b86feea 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,4 +1,4 @@
-2014-01-07  Leif Lindholm <leif.lindholm@linaro.org>
+2014-11-07  Leif Lindholm <leif.lindholm@linaro.org>
 
 	* grub-core/kern/efi/init.c: check value of *path before
 	dereferencing.
-- 
2.4.3

