From 11eed6abce6626edecdbc81876bc7f5b91458c0b Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Mon, 26 Jan 2015 09:56:24 +0100
Subject: [PATCH 241/506] util/misc.c: Check ftello return value.

Found by: Coverity scan.
---
 util/misc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/util/misc.c b/util/misc.c
index b0e023e..b8ec691 100644
--- a/util/misc.c
+++ b/util/misc.c
@@ -89,6 +89,8 @@ grub_util_get_image_size (const char *path)
   fseeko (f, 0, SEEK_END);
   
   sz = ftello (f);
+  if (sz < 0)
+    grub_util_error (_("cannot open `%s': %s"), path, strerror (errno));
   if (sz != (size_t) sz)
     grub_util_error (_("file `%s' is too big"), path);
   ret = (size_t) sz;
-- 
2.4.3

