From 3900726fa8a794869c42a0a33fbd96b5cd4ba14f Mon Sep 17 00:00:00 2001
From: Andrei Borzenkov <arvidjaar@gmail.com>
Date: Tue, 27 Jan 2015 21:13:10 +0300
Subject: [PATCH 264/506] fs/zfs/zfscrypt.c: fix indentation.

---
 grub-core/fs/zfs/zfscrypt.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/grub-core/fs/zfs/zfscrypt.c b/grub-core/fs/zfs/zfscrypt.c
index 834f858..87eef62 100644
--- a/grub-core/fs/zfs/zfscrypt.c
+++ b/grub-core/fs/zfs/zfscrypt.c
@@ -397,11 +397,11 @@ grub_zfs_load_key_real (const struct grub_zfs_key *key,
       err = grub_crypto_cipher_set_key (ret, decrypted, keylen);
       if (err)
 	{
-	    grub_errno = GRUB_ERR_NONE;
-	    grub_crypto_cipher_close (ret);
-	    grub_crypto_cipher_close (cipher);
-	    continue;
-	  }
+	  grub_errno = GRUB_ERR_NONE;
+	  grub_crypto_cipher_close (ret);
+	  grub_crypto_cipher_close (cipher);
+	  continue;
+	}
       grub_crypto_cipher_close (cipher);
       return ret;
     }
-- 
2.4.3

