From 1a4e47879c55390bd80e6f06bee68317b4f16cfa Mon Sep 17 00:00:00 2001
From: Andrei Borzenkov <arvidjaar@gmail.com>
Date: Tue, 27 Jan 2015 22:34:50 +0300
Subject: [PATCH 268/506] Change quotes to match overall style in NEWS

---
 NEWS | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/NEWS b/NEWS
index 55f09d6..e417eba 100644
--- a/NEWS
+++ b/NEWS
@@ -85,7 +85,7 @@ New in 2.02:
   * Support for USB debug dongles.
   * Support for *-emu on all platforms (previously only i386/x86_64 worked).
   * Support *-emu on Windows.
-  * New platform "none" which builds only user level utilities. This is now
+  * New platform `none' which builds only user level utilities. This is now
     default if target CPU is not supported.
 
 * Security:
-- 
2.4.3

