From 251ddf26dc80d1fa91ca3f242da8613bc59c930d Mon Sep 17 00:00:00 2001
From: Andrei Borzenkov <arvidjaar@gmail.com>
Date: Mon, 16 Feb 2015 20:31:22 +0300
Subject: [PATCH 291/506] build-sys: add syslinux test files to tarball

---
 conf/Makefile.extra-dist | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/conf/Makefile.extra-dist b/conf/Makefile.extra-dist
index 1b95ccd..8a65e37 100644
--- a/conf/Makefile.extra-dist
+++ b/conf/Makefile.extra-dist
@@ -121,3 +121,15 @@ EXTRA_DIST += tests/file_filter/file.xz.sig
 EXTRA_DIST += tests/file_filter/keys
 EXTRA_DIST += tests/file_filter/keys.pub
 EXTRA_DIST += tests/file_filter/test.cfg
+EXTRA_DIST += tests/syslinux/ubuntu10.04/isolinux/prompt.cfg
+EXTRA_DIST += tests/syslinux/ubuntu10.04/isolinux/gfxboot.cfg
+EXTRA_DIST += tests/syslinux/ubuntu10.04/isolinux/adtxt.cfg
+EXTRA_DIST += tests/syslinux/ubuntu10.04/isolinux/isolinux.cfg
+EXTRA_DIST += tests/syslinux/ubuntu10.04/isolinux/exithelp.cfg
+EXTRA_DIST += tests/syslinux/ubuntu10.04/isolinux/txt.cfg
+EXTRA_DIST += tests/syslinux/ubuntu10.04/isolinux/menu.cfg
+EXTRA_DIST += tests/syslinux/ubuntu10.04/isolinux/stdmenu.cfg
+EXTRA_DIST += tests/syslinux/ubuntu10.04/isolinux/dtmenu.cfg
+EXTRA_DIST += tests/syslinux/ubuntu10.04/isolinux/po4a.cfg
+EXTRA_DIST += tests/syslinux/ubuntu10.04/isolinux/rqtxt.cfg
+EXTRA_DIST += tests/syslinux/ubuntu10.04_grub.cfg
-- 
2.4.3

