From 0b02bfa034be0102507fd1df56d577c9c1892df1 Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Sat, 21 Feb 2015 16:22:48 +0100
Subject: [PATCH 306/506] INSTALL: clarify that clang support is experimental

---
 INSTALL | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/INSTALL b/INSTALL
index 05cc443..afbb660 100644
--- a/INSTALL
+++ b/INSTALL
@@ -13,6 +13,9 @@ configuring the GRUB.
 
 * GCC 4.1.3 or later
   Note: older versions may work but support is limited
+
+  Experimental support for clang 3.3 or later (results in much bigger binaries)
+  for i386, x86_64, arm (except thumb), arm64, mips(el), powerpc
   Note: clang 3.2 or later works for i386 and x86_64 targets but results in
         much bigger binaries.
 	earlier versions not tested
-- 
2.4.3

