From e77dc3d9a0b672d933f328fce4332a61a9f5f707 Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Sat, 21 Feb 2015 20:24:59 +0100
Subject: [PATCH 318/506] kernel-8086: Switch to more portable .org.

---
 grub-core/tests/boot/kernel-8086.S | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/grub-core/tests/boot/kernel-8086.S b/grub-core/tests/boot/kernel-8086.S
index 510897c..5ec5368 100644
--- a/grub-core/tests/boot/kernel-8086.S
+++ b/grub-core/tests/boot/kernel-8086.S
@@ -46,5 +46,5 @@ message:
 	.ascii "Boot Test Passed Successfully\n" SUCCESSFUL_BOOT_STRING "\n"
 	.byte 0
 
-	. = base + 510
-	.short 0xaa55
\ No newline at end of file
+	.org 510
+	.short 0xaa55
-- 
2.4.3

