From afd6b6bbaedfd7724e6ccc0401c76021a429779b Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Sat, 28 Feb 2015 16:22:46 +0100
Subject: [PATCH 345/506] grub-probe: Mark a "[default=]" for translation.

---
 util/grub-probe.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/util/grub-probe.c b/util/grub-probe.c
index 8a0c4f3..08d477f 100644
--- a/util/grub-probe.c
+++ b/util/grub-probe.c
@@ -728,10 +728,12 @@ help_filter (int key, const char *text, void *input __attribute__ ((unused)))
 
       case 't':
 	{
-	  char *ret, *t = get_targets_string ();
+	  char *ret, *t = get_targets_string (), *def;
 
-	  ret = xasprintf ("%s\n%s %s [default=%s]", _("print TARGET"),
-			    _("available targets:"), t, targets[print]);
+	  def = xasprintf (_("[default=%s]"), targets[print]);
+
+	  ret = xasprintf ("%s\n%s %s %s", _("print TARGET"),
+			    _("available targets:"), t, def);
 	  free (t);
 	  return ret;
 	}
-- 
2.4.3

