From 018f79da6f6f0aab34ce501e62db77c89a23aca1 Mon Sep 17 00:00:00 2001
From: Andrei Borzenkov <arvidjaar@gmail.com>
Date: Sat, 28 Feb 2015 20:19:57 +0300
Subject: [PATCH 347/506] grub-probe: free temporary variable

---
 util/grub-probe.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/util/grub-probe.c b/util/grub-probe.c
index 08d477f..8f4aa61 100644
--- a/util/grub-probe.c
+++ b/util/grub-probe.c
@@ -735,6 +735,7 @@ help_filter (int key, const char *text, void *input __attribute__ ((unused)))
 	  ret = xasprintf ("%s\n%s %s %s", _("print TARGET"),
 			    _("available targets:"), t, def);
 	  free (t);
+	  free (def);
 	  return ret;
 	}
 
-- 
2.4.3

