From 9c07daaf917ddbcb738233aa1d3a209b179f3444 Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Fri, 20 Mar 2015 11:13:58 +0100
Subject: [PATCH 365/506] cacheinfo: Add missing license information.

---
 grub-core/commands/cacheinfo.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/grub-core/commands/cacheinfo.c b/grub-core/commands/cacheinfo.c
index ead6ff8..d34a346 100644
--- a/grub-core/commands/cacheinfo.c
+++ b/grub-core/commands/cacheinfo.c
@@ -23,6 +23,8 @@
 #include <grub/i18n.h>
 #include <grub/disk.h>
 
+GRUB_MOD_LICENSE ("GPLv3+");
+
 static grub_err_t
 grub_rescue_cmd_info (struct grub_command *cmd __attribute__ ((unused)),
     int argc __attribute__ ((unused)),
-- 
2.4.3

