From c14f8a9366517270c2171757e1b9a13fe0205dd3 Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Fri, 20 Mar 2015 21:42:48 +0100
Subject: [PATCH 373/506] modinfo.sh.in: Add missing config variables.

---
 grub-core/modinfo.sh.in | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/grub-core/modinfo.sh.in b/grub-core/modinfo.sh.in
index 09d22f9..faf0ad3 100644
--- a/grub-core/modinfo.sh.in
+++ b/grub-core/modinfo.sh.in
@@ -19,11 +19,17 @@ grub_target_cflags='@TARGET_CFLAGS@'
 grub_target_cppflags='@TARGET_CPPFLAGS@'
 grub_target_ccasflags='@TARGET_CCASFLAGS@'
 grub_target_ldflags='@TARGET_LDFLAGS@'
+grub_cflags='@CFLAGS@'
+grub_cppflags='@CPPFLAGS@'
+grub_ccasflags='@CCASFLAGS@'
+grub_ldflags='@LDFLAGS@'
 grub_target_strip='@TARGET_STRIP@'
 grub_target_nm='@TARGET_NM@'
 grub_target_ranlib='@TARGET_RANLIB@'
 grub_target_objconf='@TARGET_OBJCONV@'
 grub_target_obj2elf='@TARGET_OBJ2ELF@'
+grub_target_img_base_ldopt='@TARGET_IMG_BASE_LDOPT@'
+grub_target_img_ldflags='@TARGET_IMG_BASE_LDFLAGS@'
 
 # Version
 grub_version="@VERSION@"
-- 
2.4.3

