From 2dcf260d899f65cd84e4f676e03d76f51ae01736 Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Thu, 7 May 2015 19:45:06 +0200
Subject: [PATCH 408/506] util/grub-mkrescue: Fix compilation

---
 util/grub-mkrescue.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/grub-mkrescue.c b/util/grub-mkrescue.c
index 5a7993f..4511826 100644
--- a/util/grub-mkrescue.c
+++ b/util/grub-mkrescue.c
@@ -452,7 +452,7 @@ main (int argc, char *argv[])
     for (i = 1; i < argc; i++)
       {
 	if (strcmp (argv[i], "-output") == 0) {
-	  argp_argv[argp_argc++] = "--output";
+	  argp_argv[argp_argc++] = (char *) "--output";
 	  i++;
 	  argp_argv[argp_argc++] = argv[i];
 	  continue;
-- 
2.4.3

