From 697eceff6a168e8712fb7b784b333e0aeb925958 Mon Sep 17 00:00:00 2001
From: Toomas Soome <tsoome@me.com>
Date: Thu, 4 Jun 2015 19:41:54 +0300
Subject: [PATCH 426/506] multiboot_header_tag_module_align fix to confirm
 multiboot specification

---
 include/multiboot2.h | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/include/multiboot2.h b/include/multiboot2.h
index 3ccff15..9d48627 100644
--- a/include/multiboot2.h
+++ b/include/multiboot2.h
@@ -156,9 +156,6 @@ struct multiboot_header_tag_module_align
   multiboot_uint16_t type;
   multiboot_uint16_t flags;
   multiboot_uint32_t size;
-  multiboot_uint32_t width;
-  multiboot_uint32_t height;
-  multiboot_uint32_t depth;
 };
 
 struct multiboot_color
-- 
2.4.3

