From ede25cf0ee4a6078f21d580f10d62de238b1729b Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 22 Jul 2015 16:09:59 -0400
Subject: [PATCH 10006/10006] Try it in gentpl again...

---
 grub-core/Makefile.core.def | 22 +++++++++++++++++++++-
 1 file changed, 21 insertions(+), 1 deletion(-)

diff --git a/grub-core/Makefile.core.def b/grub-core/Makefile.core.def
index 9fa9790..eed40d2 100644
--- a/grub-core/Makefile.core.def
+++ b/grub-core/Makefile.core.def
@@ -44,6 +44,18 @@ transform_data = {
   enable = powerpc_ieee1275;
 };
 
+library = {
+  name = kernel_noemu_nodist_symlist.a;
+  noemu_nodist = symlist.c;
+  ia64_efi_cflags = '-fno-builtin -fpic -minline-int-divide-max-throughput -g0';
+  x86_64_xen_cppflags	   = '$(CPPFLAGS_XEN) -g0';
+  i386_xen_cppflags	   = '$(CPPFLAGS_XEN) -g0';
+  mips_arc_cppflags = '-DGRUB_DECOMPRESSOR_LINK_ADDR=$(TARGET_DECOMPRESSOR_LINK_ADDR) -g0';
+  i386_qemu_cppflags     = '-DGRUB_BOOT_MACHINE_LINK_ADDR=$(GRUB_BOOT_MACHINE_LINK_ADDR) -g0';
+  emu_cflags = '$(CFLAGS_GNULIB) -g0';
+  emu_cppflags = '$(CPPFLAGS_GNULIB) -g0';
+};
+
 kernel = {
   name = kernel;
 
@@ -133,7 +145,7 @@ kernel = {
   noemu = kern/time.c;
   noemu = kern/generic/millisleep.c;
 
-  noemu_nodist = symlist.c;
+  ldadd = kernel_noemu_nodist_symlist.a;
 
   mips = kern/generic/rtc_get_time_ms.c;
 
@@ -350,6 +362,13 @@ program = {
   enable = emu;
 };
 
+library = {
+  name = emu_nodist_symlist.a;
+  emu_nodist = symlist.c;
+  cflags = '-g0';
+  cppflags = '-g0';
+};
+
 program = {
   name = grub-emu-lite;
 
@@ -357,6 +376,7 @@ program = {
   emu_nodist = symlist.c;
 
   ldadd = 'kernel.exec$(EXEEXT)';
+  ldadd = emu_nodist_symlist.a;
   ldadd = 'gnulib/libgnu.a $(LIBINTL) $(LIBUTIL) $(LIBSDL) $(LIBUSB) $(LIBPCIACCESS) $(LIBDEVMAPPER) $(LIBZFS) $(LIBNVPAIR) $(LIBGEOM)';
 
   enable = emu;
-- 
2.5.0

