From: Valdimir Serbinenko <phcoder@gmail.com>
Subject: Check for ieee1275 and not ieee1276.

upstream rev 4404

--- a/util/grub-install.in	2012-05-31 11:38:21 +0000
+++ b/util/grub-install.in	2012-06-01 20:43:10 +0000
@@ -738,7 +738,7 @@
 elif [ "${grub_modinfo_target_cpu}-${grub_modinfo_platform}" = "i386-ieee1275" ] || [ "${grub_modinfo_target_cpu}-${grub_modinfo_platform}" = "powerpc-ieee1275" ]; then
 
     # If a install device is defined, copy the core.elf to PReP partition.
-    if [ "${grub_modinfo_target_cpu}-${grub_modinfo_platform}" = "powerpc-ieee1276" ] && [ -n "${install_device}" ]; then
+    if [ "${grub_modinfo_target_cpu}-${grub_modinfo_platform}" = "powerpc-ieee1275" ] && [ -n "${install_device}" ]; then
         if [ "$("${grub_probe}" -m "${device_map}" -d "${install_device}" -t msdos_parttype)" != "41" ]; then
 	       gettext "The chosen partition is not a PReP partition." 1>&2
 	       echo 1>&2
